### DISCUSSION TITLE
Terminology: `Copy format for new child nodes` is not precise

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#657] [General]
The label `Copy format for new child nodes` is not precise, because what gets copied is the format AND style. In other words, it applies the same style, and also the formatting settings of the parent node that are different from the style. The same goes for `Copy format for new sibling nodes`.

The complete description would be: Copy style and format for new child nodes

So, some ideas of label as replacement (trying to make it as short as possible):

1. Copy style and format for new child
2. Keep style and format on new child
4. New child keeps style and format
5. Inherit parent style and format
6. Child inherits style and format
7. New child inherits
8. Inheritance

**Comment by macmarrum** [comment]
A good point :+1: 

I vote for 
- New sibling inherits style and format
- New child inherits style and format

**Reply by dpolivaev** [reply]
Maybe "new childen/siblings get" instead of "inherits"?

**Reply by euu2021** [reply]
Yes, I noticed that the word "inherits" can cause confusion, because people usually think of inheritance as a relation between parent and child. So, when reading the two options:

- New child inherits style and format
- New sibling inherits style and format

One may think: 

- The first one means: if I create a new child **_from the parent_**, it will inherit from the parent
- The second one means: complementing the first option, if I create a new node **_from the sibling_**, the new node will inherit from the parent, too

So, using the word "get" sounds like a good option.

**Reply by dpolivaev** [reply]
Actually, I don't think so.

Compare with this suggestions:

| Old | Dimitry | Mac |
|-|-|-|
| Apply style | Style | Style |
| Apply level styles | | Automatic level styles |
| Set new node style | Preset style for new nodes | Style for new nodes |
| Copy format to new sibling nodes | Preset format for new sibling nodes | Copy format to new sibling nodes |
| Copy format to new child nodes | Preset format for new child nodes | Copy format to new child nodes |

This action has a double function. If no the action check mark is not set, using it stores the format of the currently selected node and activates the rule. If the action check mark is set, using it again deletes the stored information and deactivates the rule.

Therefore I suggested to use word "preset". I did not want to have a debate and accepted "copy". But neither "inherits" nor "gets" gives you any clue about it.

If you ask me I still think that "preset" is the best name and description from all suggestions.

**Reply by macmarrum** [reply]
> This action has a double function. If no the action check mark is not set, using it stores the format of the currently selected node and activates the rule. If the action check mark is set, using it again deletes the stored information and deactivates the rule.

The style isn't "stored" at the time of option activation but at the time of creating a new node. Style and format is copied from the originating node.

I agree that "inherits" might be misunderstood. As mentioned before, "preset" feels a bit off. To _preset_ is to know in advance, which isn't the case here. 

I don't have any better ideas. Let's continue with the current names.

**Reply by dpolivaev** [reply]
Maybe "preserve", if it is copied from the selection before the node creation?

**Comment by macmarrum** [comment]
Thanks for the idea :+1: 

At first sight it looks good:
* Preserve style and format for new sibling nodes
* Preserve style and format for new child nodes

**Reply by euu2021** [reply]
I wasn't sure about the grammar, but it seems that "child nodes" is indeed correct (instead of "children nodes"). It has twice the number of results on google and is used by big companies.

It's funny that when I search for "child as adjective" on google, one of the first results is [a discussion about](https://english.stackexchange.com/questions/468019/children-as-adjective-as-opposed-to-child) "child nodes" in the computer science context.
